### PR TITLE
Fix the NullPointerException when "start param to overwrite global param"

### LINKED ISSUE #15676: [Bug] [Master] Create WorkflowExecuteRunnable failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

[WI-0][TI-0] - [ERROR] 2024-03-06 08:52:31.137 +0800 o.a.d.s.m.r.MasterSchedulerBootstrap:[156] - Master handle command 2007 error 
org.apache.dolphinscheduler.server.master.exception.WorkflowCreateException: Create WorkflowExecuteRunnable failed
	at org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteRunnableFactory.createWorkflowExecuteRunnable(WorkflowExecuteRunnableFactory.java:98)
	at org.apache.dolphinscheduler.server.master.runner.MasterSchedulerBootstrap.lambda$run$0(MasterSchedulerBootstrap.java:139)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
	at org.apache.dolphinscheduler.server.master.runner.MasterSchedulerBootstrap.run(MasterSchedulerBootstrap.java:136)
Caused by: java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.setGlobalParamIfCommanded(ProcessServiceImpl.java:604)
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.generateNewProcessInstance(ProcessServiceImpl.java:561)
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.constructProcessInstance(ProcessServiceImpl.java:707)
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.handleCommand(ProcessServiceImpl.java:291)

### What you expected to happen

在执行“set start param into global params”逻辑时，需要判空处理

image

### How to reproduce

工作流定义一个全局参数，类型为OUT，值为空，然后正常执行任务

image

### Anything else

_No response_

### Version

3.2.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wustlz**
## Purpose of the pull request

This pull request to fix #15676 , fix the NullPointerException by checking whether the key exists when "start param to overwrite global param"

## Brief change log

Add a "containsKey" judgement when "start param to overwrite global param"

## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

**Comment by rickchengx** [comment]
@wustlz please rebase your PR and link this to an issue

**Comment by wustlz** [comment]
> @wustlz please rebase your PR and link this to an issue

done

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 1]
```diff
@@ -601,9 +601,11 @@ public void setGlobalParamIfCommanded(ProcessDefinition processDefinition, Map<S
...
             // start param to overwrite global param
             for (Map.Entry<String, String> param : globalMap.entrySet()) {
-                String val = startParamMap.get(param.getKey()).getValue();
-                if (val != null) {
-                    param.setValue(val);
+                if (startParamMap.containsKey(param.getKey())) {
+                    String val = startParamMap.get(param.getKey()).getValue();
+                    if (val != null) {
+                        param.setValue(val);
+                    }
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: Thanks for your PR, please add UT for this case.

Comment (wustlz) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: done

**Review by rickchengx — COMMENTED** [review]
@wustlz overall LGTM, please Run 'mvn spotless:apply' to fix the code style

**Comment by wustlz** [comment]
> @wustlz overall LGTM, please Run 'mvn spotless:apply' to fix the code style

**Review by rickchengx — APPROVED** [review]
LGTM

**Pull request merged**